docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
```

Comment:
SG
What about the service account? There are additional steps needed, but the same code still works.

Do you think of any cases that developers need to explicitly create credentials?